### PR TITLE
build+htlcswitch+lnwallet: add prefix logger

### PR DESCRIPTION
**Pull request by joostjager**
This pr unifies logging statements in `link` and `channel` via a prefix logger.

### CODE REVIEW

File: build/prefix_log.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+// Criticalf formats message according to format specifier and writes to log
+// with LevelCritical.
+func (p *PrefixLog) Criticalf(format string, params ...interface{}) {
+	p.log.Criticalf(p.addFormatPrefix(format), params...)
+}
+
+// Trace formats message using the default formats for its operands and writes
+// to log with LevelTrace.
+func (p *PrefixLog) Trace(v ...interface{}) {
+
```
Comment (cfromknecht) [inline build/prefix_log.go]: nit: extra newline

File: htlcswitch/link.go [hunk 2]
```diff
@@ -642,9 +648,7 @@ func (l *channelLink) syncChanStates() error {
 			localChanSyncMsg.NextLocalCommitHeight == 1 &&
 			!l.channel.IsPending() {
 
-			log.Infof("ChannelPoint(%v): resending "+
-				"FundingLocked message to peer",
-				l.channel.ChannelPoint())
+			l.log.Infof("resending FundingLocked message to peer")
```
Comment (cfromknecht) [inline htlcswitch/link.go]: do we have a preference as to whether these should capitalized or not? i see that we are doing both atm. maybe just something to keep in mind when adding new ones

Comment (joostjager) [reply htlcswitch/link.go]: I converted everything to lower case. I found the format `ChannelLink(1:2:3): line starts lower case` looked best.

**Comment by joostjager** [comment]
Open question is what we think is the best prefix to use for link and channel? uint64, shortchan, chanpoint?

File: build/prefix_log.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import "github.com/btcsuite/btclog"
+
+// PrefixLog is a pass-through logger that adds a prefix to every logged line.
+type PrefixLog struct {
+	log    btclog.Logger
+	prefix string
+}
+
+var _ btclog.Logger = &PrefixLog{}
```
[the remark this answers is not in the document]
Comment (joostjager) [reply build/prefix_log.go]: Comment added. Check interface

File: build/prefix_log.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	p.log.Critical(p.addArgsPrefix(v)...)
+}
+
+// Level returns the current logging level.
+func (p *PrefixLog) Level() btclog.Level {
+	return p.log.Level()
+}
+
+// SetLevel changes the logging level to the passed level.
+func (p *PrefixLog) SetLevel(level btclog.Level) {}
```
[the remark this answers is not in the document]
Comment (joostjager) [reply build/prefix_log.go]: oops, fixed

[the remark this answers is not in the document]
Comment (joostjager) [reply build/prefix_log.go]: although this probably never called.

**Comment by joostjager** [comment]
>Open question is what we think is the best prefix to use for link and channel? uint64, shortchan, chanpoint?

Left formats to what they originally were (link: shortchan, channel: chanpoint).

**Pull request merged**